{
  signallingHandler.uncaughtException(anyObject(Thread.class),eq(EXCEPTION));
  Runnable runnable=createMock(Runnable.class);
  runnable.run();
  expectLastCall().andThrow(EXCEPTION);
  control.replay();
  try {
    executorService.submit(runnable).get();
    fail(EXCEPTION.getClass().getSimpleName() + " should be thrown.");
  }
 catch (  ExecutionException e) {
    assertEquals(EXCEPTION,e.getCause());
  }
}
